sketchlib/Grid.js

```diff
@@ -257,6 +257,31 @@ export class Grid {
     return index.down();
   }
 
+  /**
+   * Get a neighbor from the grid.
+   * @param {Index2D} index the index to check
+   * @param {Direction} direction The direction to the adjacent cell
+   * @returns {Index2D | undefined} The neighbor, or undefined if at the edge
+   * of the grid.
+   */
+  get_neighbor(index, direction) {
```

Comment:
Add tests for this

Reply:
Actually... Do I want this to replace the existing right/up/left/down functions? in which case I could just edit the existing tests. I need to chew on this. Leave it for now.